### ISSUE TITLE
Request for more CNavArea functions

### ISSUE THREAD

**Issue by ghost** [Facepunch/garrysmod-requests#1122]
I'm unable to list these all, but there are some unused variables of the CNavArea that could potentially be useful. All of these are seemingly used in Left 4 Dead, and one of them I'm requesting is from CS:S, but was removed.

m_potentiallyVisibleAreas -> This is a list of areas you can potentially see from the specified area. This could be absolutely revolutionary in realistic combat systems for AI/finding cover, and all the expensive computation is done while the NavMesh is being generated.
m_spotEncounters -> These spots are, in Valve's own words: "list of possible ways to move thru this area, and the spots to look at as we do"

Next up is an old feature from CS:S's navmesh system: approach areas, which got removed for some reason. The relevant code for it is [https://github.com/VSES/SourceEngine2007/blob/43a5c90a5ada1e69ca044595383be67f40b33c61/se2007/game/server/nav_generate.cpp#L89](here) and [https://github.com/VSES/SourceEngine2007/blob/master/se2007/game/server/nav_generate.cpp#L1771](here).

From Valve: 
* An approach area is an area representing a place where players 
 * move into/out of our local neighborhood of areas.

These functions would work similar to the GetHidingSpots function and are computed in a similar way, so hopefully they won't be hard to add. It'd also be nice if we could get "count" functions so we wouldn't have to get the length of the table each time.

EDIT: CNavArea::MakeNewMarker would also be a useful binding for doing certain pathing searches.

EDIT2: If these are too much to ask for, just GetSpotEncounters would do.

**Comment by T-Rizzle12** [comment]
I agree with this. Especially since the game seems to already generate encounters spots during nav_generation.

**Comment by robotboy655** [comment]
I have added m_potentiallyVisibleAreas and m_spotEncounters bindings to `dev` beta.
